File: apps/api/src/ONENET.Application/Common/Models/ApiResponse.cs
// QUAN-20260530-2301
namespace ONENET.Application.Common.Models;

public class ApiResponse
{
    public bool Success { get; set; }
    public string? Message { get; set; }
    public List<ApiError>? Errors { get; set; }

    public static ApiResponse SuccessResult(string? message = null) => new() { Success = true, Message = message };
    public static ApiResponse<T> SuccessResult<T>(T data, string? message = null) => new() { Success = true, Data = data, Message = message };
    public static ApiResponse ErrorResult(string? message = null, List<ApiError>? errors = null) => new() { Success = false, Message = message, Errors = errors };
    public static ApiResponse<T> ErrorResult<T>(string? message = null, List<ApiError>? errors = null) => new() { Success = false, Message = message, Errors = errors };
}

public class ApiResponse<T> : ApiResponse
{
    public T? Data { get; set; }
}

public class ApiError
{
    public string? Field { get; set; }
    public string? Message { get; set; }
    public string? Code { get; set; } // For specific error codes like STUDENT_HAS_RELATED_DATA
}

File: apps/web/src/types/ApiResponse.ts
// QUAN-20260530-2301
export interface ApiError {
  field?: string;
  message?: string;
  code?: string;
}

export interface ApiResponse<T = any> {
  success: boolean;
  message?: string;
  data?: T;
  errors?: ApiError[];
}

export interface PaginatedResponse<T> {
  items: T[];
  pageNumber: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}